sechub-administration/src/test/java/com/daimler/sechub/domain/administration/project/ProjectAdministrationRestControllerMockTest.java

```diff
@@ -162,6 +178,38 @@ public void delete_project_calls_delete_service() throws Exception {
         verify(projectDeleteService).deleteProject("projectId1");
     }
 
+    @Test
+    public void change_project_calls_change_details() throws Exception {
+
+        /* execute + test @formatter:off */
+        
+        this.mockMvc.perform(
+                put(https(PORT_USED).buildAdminChangesProjectDescriptionUrl(PROJECT_ID.pathElement()), "project1").
+                contentType(MediaType.APPLICATION_JSON_VALUE).
+                accept(MediaType.APPLICATION_JSON_VALUE).
+                content("{\"description\":\"new description\"}")
```

Comment:
I would rather not introduce `PATCH` into SecHub, because at the moment it is not used in the code. `PUT` on the other hand is used at least thirteen times.

At the moment only changing the `description` field was the requirement, but the same endpoint should be extended to change all project related fields in one request in the future.

Also I think `PATCH` has kind of a meaning of "correct something poorly or provisional".

Reply:
It is okay if you do not use `PATCH` and use `PUT`. However, `PUT` requires you to transfer the entire "object". You cannot just pass `content("{\"description\":\"new description\"}")`. You would need to pass the entire object in the request: https://tools.ietf.org/html/rfc2068#section-9.6 and https://tools.ietf.org/html/rfc2068#section-19.6.1.1.

Furthermore, because it should be possible to change other fields as well in the future, then the URL should be more generic. Instead of `buildAdminChangesProjectDescriptionUrl` the URL should be `buildAdminChangesProjectUrl`.

Reply:
You decided to go with `PUT`, but that would require to send the entire project information. Not just a delta of what you want to change.